aip/general/0162.md

```diff
@@ -102,57 +105,44 @@ revision. The content of `publishers/{publisher}/books/{book}/revisions/latest`
 and `publishers/{publisher}/books/{book}` can differ, as the latest revision may
 be different from the current state of the resource.
 
-### Tagging
-
-Rather than create a revision via the standard method, it may be more convenient
-to create a revision snapshotting the current resource configuration.
+### User-Specified Aliases
 
-APIs **may** provide a mechanism for users to use the existing revisions or
-current resource state to create a new revision with a custom method "tag":
+APIs **may** provide a mechanism for users to assign an [alias][] ID to an
+existing revision with a custom method "alias":
 
 ```proto
-rpc TagBookRevision(TagBookRevisionRequest) returns (Book) {
+rpc AliasBookRevision(TagBookRevisionRequest) returns (Book) {
   option (google.api.http) = {
-    post: "/v1/{name=publishers/*/books/*}/revisions:tag"
+    post: "/v1/{name=publishers/*/books/*/revisions/*}:alias"
     body: "*"
   };
 }
 ```
 
 ```proto
-message TagBookRevisionRequest {
+message AliasBookRevisionRequest {
   string name = 1 [
     (google.api.field_behavior) = REQUIRED,
     (google.api.resource_reference) = {
-      type: "library.googleapis.com/Book"
+      type: "library.googleapis.com/BookRevision"
     }];
 
-  // If set, the specified ID will be used. Otherwise, a random
-  // ID will be provided.
-  string revision_id = 2 [(google.api.field_behavior) = OPTIONAL];
-
-  // If unset, the current resource configuration will used to create the
-  // revision.
-  //
-  // If set, the configuration in the specified revision will be used. This is
-  // helpful when one wants to set an alias for the resource.
-  string source_revision_id = 3 [(google.api.field_behavior) = OPTIONAL];
+  // The ID of the revision to alias to, e.g. `CURRENT` or a semantic
+  // version.
+  string alias_id = 2 [(google.api.field_behavior) = REQUIRED];
```

Comment:
I think we can't really do it in the `Create` call without having some sort of initial alias in the resource state, which might get stale if a different version creates itself. 

So I guess making this optional is the easiest way to do so? wand we can document the behavior as such.

For delete, I think we can add a clause that if the delete is an alias, then the alias is removed but the actual revision is not? It's a little wonky since the delete call wouldn't actually delete a resource, but in practice I can't think of any issues.

Reply:
Delete: yes, that works for me.

Create: Aargh - I've confused things by adding my comment to `alias_id` instead of `name`. `alias_id` should be required, and `name` has to be required really, because otherwise we can't identify the original resource. It's almost like we want the field to refer to "the revision or the resource" ;)
Back when the aliasing operation was on the resource itself, this was okay... but now it's not :( Basically we want three pieces of information:

- The Book resource (required)
- The "target" alias ID (required)
- The "source" revision ID (optional)

And we've only got two fields. Darn.

Maybe we can just live with this for now.

Reply:
Yup, that works for me.